packages/integrations/node/src/nodeMiddleware.ts

```diff
@@ -60,34 +62,39 @@ export default function (app: NodeApp, mode: Options['mode']) {
 	};
 }
 
-async function writeWebResponse(app: NodeApp, res: ServerResponse, webResponse: Response) {
-	const { status, headers } = webResponse;
+async function writeWebResponse(app: NodeApp, res: ServerResponse, webResponse: Response, logger: AstroIntegrationLogger) {
+	const { status, headers, body } = webResponse;
 
 	if (app.setCookieHeaders) {
 		const setCookieHeaders: Array<string> = Array.from(app.setCookieHeaders(webResponse));
 
 		if (setCookieHeaders.length) {
 			for (const setCookieHeader of setCookieHeaders) {
-				webResponse.headers.append('set-cookie', setCookieHeader);
+				headers.append('set-cookie', setCookieHeader);
 			}
 		}
 	}
 
 	const nodeHeaders = createOutgoingHttpHeaders(headers);
 	res.writeHead(status, nodeHeaders);
-	if (webResponse.body) {
+	if (body) {
 		try {
-			const reader = webResponse.body.getReader();
+			const reader = body.getReader();
 			res.on('close', () => {
-				reader.cancel();
+				// Cancelling the reader may reject not just because of
+				// an error in the ReadableStream's cancel callback, but
+				// also because of an error anywhere in the stream.
+				reader.cancel().catch(err => {
+					logger.error(`There was an uncaught error in the middle of the stream while rendering ${res.req.url}.`);
+					console.error(err);
+				});
 			});
 			let result = await reader.read();
 			while (!result.done) {
 				res.write(result.value);
 				result = await reader.read();
 			}
 		} catch (err: any) {
-			console.error(err?.stack || err?.message || String(err));
 			res.write('Internal server error');
```

Comment:
This could now potentially swallow errors without logging. Should keep the logging here as before?

Reply:
That would end up logging twice, because stream errors are thrown on cancelling the reader.

Reply:
I guess I was worried that `res.on('close'` isn't always guaranteed to be called, which means `reader.cancel()` may not be called. So it feels safer to log it in this catch block instead. Maybe we can do `reader.cancel().catch(() => {})` instead since we know the error should already be logged here?

Reply:
Wouldn't `res.end()` below this catch block gurantee that?

Reply:
True. In that case, maybe we should leave a comment here instead about why we're swallowing the error (also remove the `(err: any)` part)